tests/basic/src/test/java/org/jboss/hal/testsuite/test/runtime/KillDestroyServerGroupsTest.java

```diff
@@ -0,0 +1,119 @@
+import org.jboss.arquillian.drone.api.annotation.Drone;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.Property;
+import org.jboss.hal.meta.token.NameTokens;
+import org.jboss.hal.resources.Ids;
+import org.jboss.hal.testsuite.Console;
+import org.jboss.hal.testsuite.category.Domain;
+import org.jboss.hal.testsuite.creaper.ManagementClientProvider;
+import org.jboss.hal.testsuite.fragment.finder.FinderPath;
+import org.jboss.hal.testsuite.util.ConfigUtils;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.openqa.selenium.WebDriver;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+import org.wildfly.extras.creaper.core.online.operations.Values;
+import org.wildfly.extras.creaper.core.online.operations.admin.DomainAdministration;
+
+@RunWith(Arquillian.class)
+@Category(Domain.class)
+public class KillDestroyServerGroupsTest {
+
+    private static final OnlineManagementClient client = ManagementClientProvider.createOnlineManagementClient();
+    private static final DomainAdministration domainAdministration = new DomainAdministration(client);
+    private static final Operations operations = new Operations(client);
+    private static final FinderPath SERVER_GROUP_FINDER_PATH =
+        new FinderPath().append(Ids.DOMAIN_BROWSE_BY, "server-groups");
+    private static final String OTHER_SERVER_GROUP = "other-server-group";
+    private static final String OTHER_SERVER_GROUP_ID = Ids.build("sg", OTHER_SERVER_GROUP);
+
+    private static String[] SERVERS_IN_OTHER_SERVER_GROUP;
+
+    @BeforeClass
+    public static void setUp() throws IOException {
+        SERVERS_IN_OTHER_SERVER_GROUP =
+            operations.invoke("read-children-resources", Address.host(ConfigUtils.getDefaultHost()),
+                Values.of("child-type", "server-config")).listValue()
+                .stream()
+                .map(ModelNode::asProperty)
+                .filter(property -> property.getValue().get("group").asString().equals(OTHER_SERVER_GROUP))
+                .map(Property::getName)
+                .toArray(String[]::new);
+    }
+
+    @AfterClass
+    public static void tearDown() throws IOException {
+        client.close();
+    }
+
+    @Drone
+    private WebDriver browser;
+
+    @Inject
+    private Console console;
+
+    @Before
+    public void startServersInServerGroupIfNeeded() throws IOException {
+        operations.invoke("start-servers",
+            Address.root().and("server-group", OTHER_SERVER_GROUP), Values.of("blocking", true));
+    }
+
+    private void verifyServersAreStopped() throws IOException, InterruptedException {
```

Comment:
Maybe you can add a parameter with the desired state so you can use this method for the 'started check' in the `startServersInServerGroupIfNeeded` method as well. WDYT?

Reply:
Fixed thanks!